server/pages/index.js

```diff
@@ -408,13 +354,33 @@ module.exports = function() {
 
 	// Isomorphic routing
 	sections
-		.filter( section => section.isomorphic )
 		.forEach( section => {
-			sectionsModule.require( section.module )( serverRouter( app, setUpRoute, section ) );
+			section.paths.forEach( path => {
+				const pathRegex = utils.pathToRegExp( path );
+
+				app.get( pathRegex, function( req, res, next ) {
+					if ( config.isEnabled( 'code-splitting' ) ) {
+						req.context = Object.assign( {}, req.context, { chunk: section.name } );
+					}
+					next();
+				} );
+
+				if ( ! section.isomorphic ) {
+					if ( section.enableLoggedOut ) {
+						app.get( pathRegex, setUpRoute, serverRender );
```

Comment:
> Asking without a fuller context: why `setUpRoute` and not `setUpLoggedOutRoute`?

The `section.enableLoggedOut` conditional corresponds to the special routes I'm removing with this PR (the ones that failed with the earlier version, i.e. `/start`, `/jetpack/connect`, etc). They were using `setUpRoute`-- the reason is we want their `context` to be set up properly when logged-in.

> (Aside: the innermost `if-else` pair can be refactored into one `app.get` call)

Good call. 

Actually considering dropping `setUpRoute` altogether, and turning `if ( section.enableLoggedOut ) {` into `if ( section.enableLoggedOut && ! req.cookies.wordpress_logged_in ) {`.